jina/optimizers/flow_runner.py

```diff
@@ -0,0 +1,88 @@
+import os
+from pathlib import Path
+import shutil
+from ruamel.yaml import YAML
+
+from jina.flow import Flow
+from jina.helper import colored
+from jina.logging import default_logger as logger
+
+
+class FlowRunner:
+    def __init__(
+        self,
+        flow_yaml,
+        documents,
+        batch_size,
+        pod_dir,
+        task,  # this can be only index or search as it is used to call the flow API
```

Comment:
task should be a parameter of run. a Flow should be described by the Pods and its parameters. what Request is sent to query is independent from it

Reply:
This will cause problem with `MultiFlowRunner`'s run as we will have to pass the `task` to it now.

Reply:
if we have MultiFlowRunner inherit itself from FlowRunner (or a common BaseClass), this would not be a problem because they would have the same interface.